packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -91,7 +91,9 @@ export async function handleHotUpdate(
 	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
 	// These will cause full reloads, so filter them out here
 	const mods = [...filtered].filter((m) => !m.url.endsWith('='));
-	const file = ctx.file.replace(config.root.pathname, '/');
+	const file = fileURLToPath(`file:///${ctx.file}`)
+		.replace(fileURLToPath(config.root), '/')
+		.replace(/\\/g, '/');
```

Comment:
Can't we make the paths relative to the Astro working directory? It seems more simple...

Reply:
That was my first attempt, I need to see if it can be simplified first using only one `fileURLToPath`